internal/yunikorn/rest_test.go

```diff
@@ -126,6 +127,616 @@ func TestRESTClient_GetApplication(t *testing.T) {
 	}
 }
 
+func TestRESTClient_GetApplications(t *testing.T) {
+	tests := []struct {
+		name      string
+		partName  string
+		queueName string
+		expected  []*dao.ApplicationDAOInfo
+	}{
+		{
+			name:      "All Fields Specified",
+			partName:  "specialPartition",
+			queueName: "myQueue",
+			expected: []*dao.ApplicationDAOInfo{
+				{
+					ApplicationID: "app1",
+					Partition:     "specialPartition",
+					QueueName:     "myQueue",
+				},
+			},
+		},
+		{
+			name:      "Empty Partition Name",
+			partName:  "",
+			queueName: "myQueue",
+			expected: []*dao.ApplicationDAOInfo{
+				{
+					ApplicationID: "app1",
+					Partition:     "default",
+					QueueName:     "myQueue",
+				},
+			},
+		},
+		{
+			name:      "Empty Queue Name",
+			partName:  "myPartition",
+			queueName: "",
+			expected: []*dao.ApplicationDAOInfo{
+				{
+					ApplicationID: "app1",
+					Partition:     "myPartition",
+					QueueName:     "myPartition.default",
+				},
+			},
+		},
+	}
+
+	partQueueAppRe := regexp.MustCompile(`/ws/v1/partition/(\w+)/queue/(\w+)/applications`)
+	partAppRe := regexp.MustCompile(`/ws/v1/partition/(\w+)/applications/active`)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				var responseApp []*dao.ApplicationDAOInfo
+
+				matches := partQueueAppRe.FindStringSubmatch(r.URL.Path)
+				if matches != nil {
+					responseApp = []*dao.ApplicationDAOInfo{
+						{
+							Partition:     matches[1],
+							QueueName:     matches[2],
+							ApplicationID: "app1",
+						},
+					}
+				} else {
+					matches = partAppRe.FindStringSubmatch(r.URL.Path)
+					if matches == nil {
+						http.Error(w, errors.New("invalid request path").Error(), http.StatusNotFound)
+						return
+					}
+
+					responseApp = []*dao.ApplicationDAOInfo{
+						{
+							Partition:     matches[1],
+							QueueName:     fmt.Sprintf("%s.%s", matches[1], "default"),
+							ApplicationID: "app1",
+						},
+					}
+
+				}
+				err := json.NewEncoder(w).Encode(&responseApp)
+				if err != nil {
+					http.Error(w, err.Error(), http.StatusInternalServerError)
+				}
+
+			}))
+
+			defer ts.Close()
+			client := NewRESTClient(getMockServerYunikornConfig(t, ts.URL))
+
+			apps, err := client.GetApplications(context.Background(), tt.partName, tt.queueName)
+			require.NoError(t, err)
```

Comment:
there is no sense doing further asserts in case of an error, all of them will fail.
```
if err != nil {
  t.Fatalf("error happened: %v", err)
}
```

Reply:
Hi, I think further asserts only execute if there is no error.
`require.NoError(t, err) // If error it exists here`

Reply:
Correct - any checks using `require` immediately stops the test (whereas `assert` checks will spew an error and keep going).

Reply:
yeah, so it's equivalent to `t.Fatalf()` anyway